components/ingest-service/pipeline/publisher/event_publisher.go

```diff
@@ -0,0 +1,118 @@
+package publisher
+
+import (
+	"context"
+	"time"
+
+	automate_event "github.com/chef/automate/api/interservice/event"
+	event "github.com/chef/automate/components/event-service/config"
+	"github.com/chef/automate/components/ingest-service/pipeline/message"
+	"github.com/gofrs/uuid"
+	"github.com/golang/protobuf/ptypes"
+	_struct "github.com/golang/protobuf/ptypes/struct"
+	log "github.com/sirupsen/logrus"
+)
+
+func BuildEventPublisher(eventServiceClient automate_event.EventServiceClient) message.ChefActionPipe {
+	return func(in <-chan message.ChefAction) <-chan message.ChefAction {
+		return eventPublisher(in, eventServiceClient)
+	}
+}
+
+func eventPublisher(in <-chan message.ChefAction,
+	eventServiceClient automate_event.EventServiceClient) <-chan message.ChefAction {
+	out := make(chan message.ChefAction, 100)
+	go func() {
+		for msg := range in {
+			if err := msg.Ctx.Err(); err != nil {
+				msg.FinishProcessing(err)
+				continue
+			}
+
+			event, err := createEvent(msg)
+			if err != nil {
+				log.Errorf("unable to send event", msg.Action.GetNodeId)
+				message.PropagateChefAction(out, &msg)
+				continue
+			}
+
+			start := time.Now()
+			req := &automate_event.PublishRequest{Msg: event}
+			_, err = eventServiceClient.Publish(context.Background(), req)
+			if err != nil {
+				log.Errorf("unable to send event", msg.Action.GetNodeId)
+			}
+			dur := time.Since(start)
+			log.WithFields(log.Fields{
+				"message_id":  msg.ID,
+				"buffer_size": len(out),
+				"dur":         dur,
+			}).Debug("Published to event-service")
+
+			// We only need to store polyicy file information.
+			if msg.InternalChefAction.EntityType != "policy" {
```

Comment:
The chef-server actions stored in the "actions-2020-07-31" indexes were only used for the event feed and policy information. Now that this information is being sent to the event-feed-service we only need to store information for policy information. Policy information is only stored in chef-server actions when the entity type equals "policy". Therefor the rest of the actions can be skipped.